public/js/actions/sources.js

```diff
@@ -184,6 +184,25 @@ function togglePrettyPrint(id) {
   };
 }
 
+function loadOriginalSourceText(source, getState, dispatch) {
+  return Task.spawn(function* () {
+    const generatedSource = getSource(
+      getState(),
+      getGeneratedSourceId(source)
+    );
+
+    const generatedSourceText = yield dispatch(
+      loadSourceText(generatedSource.toJS())
```

Comment:
Might be worth looking into, I could be wrong too. From a high-level it seems like you should be able to serve the source for a given sourcemapped URL if you have a sourcemap. I have not thought this through though!